### PR TITLE
added Instrumental to lyrics-plus, and ability to play stations with auto-shuffle on fixing #1448

### PR DESCRIPTION
**Pull request by huhridge**
added #1369

Note: Even though I have added instrumental for both Netease and musixmatch, I would recommend putting musixmatch over netease in provider order.

also added the case where if a provider only has unsynced lyrics, rest are also gone through to check if there are synced ones.

### CODE REVIEW

**Comment by afonsojramos** [comment]
Wow, thanks for the PR, I'll have to test it a bit and will get back to you soon!

**Comment by huhridge** [comment]
Please do test it, I did as much as I could, and couldn't find anything wrong, but I may have missed something

**Comment by huhridge** [comment]
Fixed: #1448
Since there is no simple way, to get the queue of a radio, added a workaround to play with original play function.

File: CustomApps/lyrics-plus/ProviderNetease.js [hunk 1]
```diff
@@ -120,6 +121,9 @@ const ProviderNetease = (function () {
         const lyrics = lines
             .map((line) => {
                 const { time, text } = parseTimestamp(line);
+                if (text === "纯音乐, 请欣赏") {
+                    isInstrumental = true;
+                }
```
Comment (afonsojramos) [inline CustomApps/lyrics-plus/ProviderNetease.js:126]: What is this exactly? 😅

**Review by afonsojramos — APPROVED** [review]
Thanks for all the fixes!! Looking good!

File: Extensions/shuffle+.js [hunk 2]
```diff
@@ -116,6 +116,10 @@
             case Spicetify.URI.Type.TRACK:
             case Spicetify.URI.Type.EPISODE:
                 return [uri];
+            case Spicetify.URI.Type.STATION:
+            case Spicetify.URI.Type.RADIO:
+                playerPlayOGFunc({ uri: uri }, { featureVersion: Spicetify.Platform.PlayerAPI._defaultFeatureVersion });
+                return ["playedstation"];
```
Comment (afonsojramos) [inline Extensions/shuffle+.js:122]: It is a workaround, but I think it is very valid.
Do we need this line though? 
```rust
playerPlayOGFunc({ uri: uri }, { featureVersion: Spicetify.Platform.PlayerAPI._defaultFeatureVersion });
```

File: CustomApps/lyrics-plus/ProviderNetease.js (hunk 1 above)
[the remark this answers is not in the document]
Comment (huhridge) [reply CustomApps/lyrics-plus/ProviderNetease.js:126]: yeah, that is the only way to detect instrumental in netease, the Musixmatch one will probably get used more often

File: Extensions/shuffle+.js (hunk 2 above)
Comment (huhridge) [reply Extensions/shuffle+.js:122]: Yeah, this is the only way to play stations, I tested it out a lot of different ones, but this is the only one that works.

**For a bit of context**: 
As I am sure you're aware, shuffle+ injects its own playing function when auto-shuffle is on, thankfully it does store the original function, which is, as the name suggests `playerPlayOGFunc`, so I have to use this to play the stations. And those arguments is the only way I know how to call it successfully.

Comment (afonsojramos) [reply Extensions/shuffle+.js:122]: I was actually not aware when I made the comment as I haven't messed around with shuffle+, but afterwards I went digging into the code and found that binding. Thanks for the context!!

**Pull request merged**